Sources/Sentry/SentrySession.m

```diff
@@ -9,61 +9,92 @@ @implementation SentrySession
 
 @synthesize flagInit = _init;
 
-- (instancetype)initWithReleaseName:(NSString *)releaseName
+/**
+ * Default private constructor.
+ */
+- (instancetype)init
 {
     if (self = [super init]) {
         _sessionId = [NSUUID UUID];
         _started = [SentryCurrentDate date];
         _status = kSentrySessionStatusOk;
         _sequence = 1;
         _errors = 0;
-        _init = @YES;
         _distinctId = [SentryInstallation id];
+    }
+
+    return self;
+}
+
+- (instancetype)initWithReleaseName:(NSString *)releaseName
+{
+    if (self = [self init]) {
+        _init = @YES;
         _releaseName = releaseName;
     }
     return self;
 }
 
 - (instancetype)initWithJSONObject:(NSDictionary *)jsonObject
 {
-    // Note this doesn't use the main init method since it should only init
-    // fields that exist in the JSON.
-    if (self = [super init]) {
-        _sessionId = [[NSUUID UUID] initWithUUIDString:[jsonObject valueForKey:@"sid"]];
-        _distinctId = [jsonObject valueForKey:@"did"];
-        NSString *startedString = [jsonObject valueForKey:@"started"];
-        if (nil != startedString) {
-            _started = [NSDate sentry_fromIso8601String:startedString];
-        }
-        NSString *timestampString = [jsonObject valueForKey:@"timestamp"];
-        if (nil != timestampString) {
-            _timestamp = [NSDate sentry_fromIso8601String:timestampString];
-        }
-        NSString *status = [jsonObject valueForKey:@"status"];
-        if ([@"ok" isEqualToString:status]) {
-            _status = kSentrySessionStatusOk;
-        } else if ([@"exited" isEqualToString:status]) {
-            _status = kSentrySessionStatusExited;
-        } else if ([@"crashed" isEqualToString:status]) {
-            _status = kSentrySessionStatusCrashed;
-        } else if ([@"abnormal" isEqualToString:status]) {
-            _status = kSentrySessionStatusAbnormal;
-        }
-        _sequence = [[jsonObject valueForKey:@"seq"] unsignedIntegerValue];
-        _errors = [[jsonObject valueForKey:@"errors"] unsignedIntegerValue];
+    // We use the default constructor here to set the non nullable values to a default values,
+    // because this could cause crashes, for example, in serialize.
+    // With this approach we avoid crashes and accept the tradeoff that some session data might not
+    // be 100% accurate.
+    if (self = [self init]) {
+        NSUUID *sessionId = [[NSUUID UUID] initWithUUIDString:[jsonObject valueForKey:@"sid"]];
+        if (nil != sessionId) {
+            _sessionId = [[NSUUID UUID] initWithUUIDString:[jsonObject valueForKey:@"sid"]];
+        }
```

Comment:
Did I read this right, if the value from json isn't null, we set to `_sessionId`? so would this do instead of reading again and creating another UUID:

```suggestion
        if (nil != sessionId) {
            _sessionId = sessionId;
        }
```